### PR TITLE
[Bugfix]: Fix crypto P-224 curve bug by updating Go version to 1.14.14

### PR DESCRIPTION
**Pull request by iholder-redhat**
Signed-off-by: Itamar Holder <iholder@redhat.com>

**What this PR does / why we need it**:
Update Go version to 1.14.14.
    
This addresses the following Bugzilla bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1929069
    
As written in bug description, this bug is fixed in Go upstream:
https://github.com/golang/go/commit/d95ca9138026cbe40e0857d76a81a16d03230871

In addition, add a small tutorial that will document the procedure needed to update Go version.
**Release note**:

```release-note
Go version updated to 1.14.14
```

### CODE REVIEW

**Comment by iholder-redhat** [comment]
/hold
This PR waits for another PR which [migrates Kubevirt to quay.io from Dockerhub](https://github.com/kubevirt/kubevirt/pull/5087).

File: docs/update-go-version.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# How to update Go version 
+A quick guide to update Kubevirt's Go version.
+
+To update the Go version we need to update the builder image so that it uses the new version,
+push it to the registry and finally let Kubevirt use the new builder image.
+
+In addition, Bazel has to be updated if the current version does not support the target Go version.
```
Comment (dhiller) [inline docs/update-go-version.md:7]:
```suggestion
In addition, [go rules for bazel](https://github.com/bazelbuild/rules_go) have to be updated if the current version does not support the target Go version.
```

File: docs/update-go-version.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* In [WORKSPACE](../WORKSPACE) change `go_version` to the desired Go version.
+  * Should look similar to:
+    ```shell
+    go_register_toolchains(
+        go_version = "1.14.14",
+        nogo = "@//:nogo_vet",
+    )
+    ```
+
+## Update Bazel Version
```
Comment (dhiller) [inline docs/update-go-version.md:39]:
```suggestion
## Update go rules for bazel
```

Comment (dhiller) [reply docs/update-go-version.md:39]: This change applies for the remaining paragraph also.

**Review by dhiller — COMMENTED** [review]
/approve

I have some nits on the update markdown file, please see them inline.

**Comment by iholder-redhat** [comment]
> /hold
> This PR waits for another PR which [migrates Kubevirt to quay.io from Dockerhub](https://github.com/kubevirt/kubevirt/pull/5087).

/hold cancel

**Review by enp0s3 — APPROVED** [review]
/lgtm

**Comment by iholder-redhat** [comment]
everything passes locally.
/retest

**Comment by iholder-redhat** [comment]
/hold
resolving conflicts

**Comment by iholder-redhat** [comment]
Waiting for https://github.com/kubevirt/kubevirt/pull/5248 to be merged.

**Comment by iholder-redhat** [comment]
> Waiting for #5248 to be merged.

merged.
/hold cancel
/test all

**Review by enp0s3 — APPROVED** [review]
/lgtm

**Pull request merged**